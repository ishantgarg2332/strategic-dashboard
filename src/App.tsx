import { FunctionComponent, Suspense, lazy } from 'react';
import { Routes, Route } from 'react-router-dom';
import { Spin } from 'antd';
import PATHS from './routes';

const LoginContainer = lazy(
  () => import(/* webpackChunkName: "Login" */ './pages/Login/Login'),
);

const SignupContainer = lazy(
  () => import(/* webpackChunkName: "Signup" */ './pages/Sign-Up/Sign-Up'),
);

const App: FunctionComponent = () => {
  return (
    <Suspense fallback={<Spin size='large' />}> 
      <Routes>
        <Route element={<LoginContainer />}
          key={PATHS.LOGIN}
          path={PATHS.LOGIN || '404'}
        />
        <Route element={<SignupContainer />}
          key={PATHS.SIGN_UP}
          path={PATHS.SIGN_UP || '404'}
        />
      </Routes>
    </Suspense>
  );
};

export default App;
