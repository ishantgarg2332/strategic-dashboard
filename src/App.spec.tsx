import { render } from '@testing-library/react';
import { LooseObject } from './globalTypes';
import App from './App';
import RouterHOC from '../__tests__/routerHOC'

describe('App', () => {
    let container: LooseObject;

    beforeAll(() => {
        const { container: wrapper } = render(
          <RouterHOC>
            <App />
          </RouterHOC>
    );

        container = wrapper;
    });

    test('if App is rendering Properly', () => {
        expect(container.firstChild).toMatchSnapshot();
        expect(container).toBeTruthy();
    })
})