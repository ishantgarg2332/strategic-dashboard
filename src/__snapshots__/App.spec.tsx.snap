// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App if App is rendering Properly 1`] = `null`;
